modules/cas_disk/exp_obj.c

```diff
@@ -574,23 +596,26 @@ int casdsk_exp_obj_create(struct casdsk_disk *dsk, const char *dev_name,
 	return 0;
 
 error_set_geometry:
+	_casdsk_exp_obj_clear_dev_t(dsk);
+error_exp_obj_set_dev_t:
 	blk_cleanup_queue(exp_obj->queue);
-error_init_queue:
+	put_disk(gd);
```

Comment:
please verify whether this call sequence is valid in case of kernel 5.15 (where gendisk and queue are both allocated using  blk_mq_alloc_disk

Reply:
cleanup function added to the configure scripts